lib/optimize/SplitChunksPlugin.js

```diff
@@ -941,6 +1117,7 @@ module.exports = class SplitChunksPlugin {
 							}
```

Comment:
`maxSize` need to be chosen as combination of `maxAsyncSize` and `maxInitialSize`.

For `chunk.isOnlyInitial()` use `maxInitialSize`.
For `chunk.canBeInitial()` use `combineSizes(maxInitialSize, maxAsyncSize, Math.min)` as we have to follow both restrictions.
For `!chunk.canBeInitial()` (else) use `maxAsyncSize`